File: src/main/java/miretz/ycloud/services/DocumentService.java
package miretz.ycloud.services;

import java.io.File;
import java.io.IOException;
import java.io.InputStream;
import java.util.List;

import miretz.ycloud.models.Document;

import com.vaadin.server.FileResource;

public interface DocumentService {

	boolean deleteFile(Document document);

	void deleteAllFiles();

	String getModifiedDate(Document document);

	FileResource getFileResource(Document document);

	FileResource getThumbnailFileResource(Document document);

	double getFreeSpace();

	double getSizeOfFiles();

	double getSizeInMbDouble(long size);

	void saveThumbnail(Document document) throws IOException;

	InputStream getAllFilesZip(List<Document> documents);

	File getFile(Document document);

	String getFileMimeType(Document document);

}

File: src/main/java/miretz/ycloud/services/DocumentService.kt
package miretz.ycloud.services

import com.vaadin.server.FileResource
import miretz.ycloud.models.Document
import java.io.File
import java.io.InputStream

interface DocumentService {

    fun deleteFile(document: Document): Boolean

    fun deleteAllFiles()

    fun getModifiedDate(document: Document): String

    fun getFileResource(document: Document): FileResource

    fun getThumbnailFileResource(document: Document): FileResource?

    fun getFreeSpace(): Double

    fun getSizeOfFiles(): Double

    fun getSizeInMbDouble(size: Long): Double

    fun saveThumbnail(document: Document)

    fun getAllFilesZip(documents: List<Document>): InputStream

    fun getFile(document: Document): File

    fun getFileMimeType(document: Document): String

}